File: Python3/convert_a_number_to_hexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        if num == 0: return "0"
        
        uNum = num if num > 0 else num + 2**32;
        result = "";
        chars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'];
        
        while uNum > 0:
            digit = uNum % 16
            result = chars[digit] + result
            uNum //= 16
        
        return result

File: C/convert_a_number_to_hexadecimal.c
char * toHex(int num){
    char* result = malloc(10 * sizeof(char));
    
    if(num == 0){
        result[0] = '0';
        result[1] = '\0';
        return result;
    }
        
    unsigned uNum = num;
    
    char chars[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
    int index = 0;
        
    while(uNum > 0){
        unsigned digit = uNum % 16;
        for(int i = index-1; i >= 0; i--){
            result[i+1] = result[i];
        }
        result[0] = chars[digit];
        uNum /= 16;
        index ++;
    }
    result[index] = '\0';
    
    return result;
}
